bottlerocket/agents/src/sonobuoy.rs

```diff
@@ -206,3 +224,101 @@ pub async fn delete_sonobuoy(kubeconfig_path: &str) -> Result<(), error::Error>
 
     Ok(())
 }
+
+/// Returns a string that should be used with the
+/// sonobuoy command line (i.e `e2e` `customPlugin.yaml`) and any extra args that should be used.
+/// If `plugin` is `Path` the string is returned.
+/// If `plugin` is `EncodedYaml` a file will be created at `/tmp/<name>` containing
+/// the decoded yaml.
+/// If `plugin` is `CustomPlugin` a file will be created at `/tmp/<name>` using
+/// `sonobuoy gen plugin` or `serde_yaml`
+pub fn create_plugin(plugin: &SonobuoyPluginConfig) -> Result<(String, Vec<String>), error::Error> {
```

Comment:
Something about calling this `create` seems off:

```suggestion
pub fn setup_plugin<P: AsRef<Path>>(plugin: &SonobuoyPluginConfig, tempdir: P) -> Result<(String, Vec<String>), error::Error> {
```